File: C#/ArraysAndHashing/ProductOfArrayExceptSelf.cs
public class Solution
{
    public int[] ProductExceptSelf(int[] nums)
    {
        int[] result = new int[nums.Length];
        int runningProduct = 1;

        for (int i = 0; i < nums.Length; i++)
        {
            result[i] = runningProduct;
            runningProduct *= nums[i];
        }

        runningProduct = 1;

        for (int i = nums.Length - 1; i >= 0; i--)
        {
            result[i] *= runningProduct;
            runningProduct *= nums[i];
        }

        return result;
    }
}

File: Python/ArraysAndHashing/ProductOfArrayExceptSelf.py
class Solution:
    def productExceptSelf(self, nums: list[int]) -> list[int]:
        result = [1] * len(nums)
        running_product = 1

        for i in range(len(nums)):
            result[i] = running_product
            running_product *= nums[i]

        running_product = 1

        for i in range(len(nums) - 1, -1, -1):
            result[i] *= running_product
            running_product *= nums[i]

        return result
